app/main.py

```diff
@@ -21,8 +22,13 @@ def main():
             print("Invalid move, please try again")
             continue
 
-        # make the move and update the board
-        chess_board.make_move(move)
+        if chess_board.is_white_piece(move) is white_turn:
```

Comment:
Instead of using the `"is"`, I find easier to use the `"!="`, i.e.:
```python
if chess_board.is_white_piece(move) != white_turn:
```
What do you think?